cli/cmd/generate_azure.go

```diff
@@ -0,0 +1,737 @@
+				return errors.Wrap(err, "failed to load command flags")
+			}
+			if err := validateOutputLocation(dirname); err != nil {
+				return err
+			}
+
+			// Validate Storage Location
+			storageLocation, err := cmd.Flags().GetString("location")
+			if err != nil {
+				return errors.Wrap(err, "failed to load command flags")
+			}
+			if err := validateStorageLocation(storageLocation); storageLocation != "" && err != nil {
+				return err
+			}
+
+			// Load any cached inputs if interactive
+			if cli.InteractiveMode() {
+				cachedOptions := &azure.GenerateAzureTfConfigurationArgs{}
+				iacParamsExpired := cli.ReadCachedAsset(CachedAzureAssetIacParams, &cachedOptions)
+				if iacParamsExpired {
+					cli.Log.Debug("loaded previously set values for Azure iac generation")
+				}
+
+				extraState := &AzureGenerateCommandExtraState{}
+				extraStateParamsExpired := cli.ReadCachedAsset(CachedAzureAssetExtraState, &extraState)
+				if extraStateParamsExpired {
+					cli.Log.Debug("loaded previously set values for Azure iac generation (extra state)")
+				}
+
+				// Determine if previously cached options exists; prompt user if they'd like to continue
+				answer := false
+				if !iacParamsExpired || !extraStateParamsExpired {
+					if err := SurveyQuestionInteractiveOnly(SurveyQuestionWithValidationArgs{
+						Prompt:   &survey.Confirm{Message: QuestionUsePreviousCache, Default: false},
+						Response: &answer,
+					}); err != nil {
+						return errors.Wrap(err, "failed to load saved options")
+					}
+				}
+
+				// If the user decides NOT to use the previous values; we won't load them.  However, every time the command runs
+				// we are going to write out new cached values, so if they run it - bail out - and run it again they'll get
+				// re-prompted.
+				if answer {
+					// Merge cached inputs to current options (current options win)
+					if err := mergo.Merge(GenerateAzureCommandState, cachedOptions); err != nil {
+						return errors.Wrap(err, "failed to load saved options")
+					}
+					if err := mergo.Merge(GenerateAzureCommandExtraState, extraState); err != nil {
+						return errors.Wrap(err, "failed to load saved options")
+					}
+				}
+			}
+
+			// Collect and/or confirm parameters
+			err = promptAzureGenerate(GenerateAzureCommandState, GenerateAzureCommandExtraState)
+			if err != nil {
+				return errors.Wrap(err, "collecting/confirming parameters")
+			}
+
+			return nil
+		},
+	}
+)
+
+func validateStorageLocation(location string) error {
+	if !validStorageLocations[location] {
+		return errors.New("invalid storage location supplied")
+	}
+	return nil
+}
+
+func initGenerateAzureTfCommandFlags() {
+	// Azure sub-command flags
+	generateAzureTfCommand.PersistentFlags().BoolVar(
+		&GenerateAzureCommandState.ActivityLog,
+		"activity_log",
+		false,
+		"enable active log integration")
+
+	generateAzureTfCommand.PersistentFlags().StringVar(
+		&GenerateAzureCommandState.ActivityLogIntegrationName,
+		"activity_name",
```

Comment:
IMO we should be verbose with these to make it's intention clear & obvious
```suggestion
		"activity_log_integration_name",
```

Reply:
Done